packages/core/src/authenticatedFetch/dpopUtils.ts

```diff
@@ -30,11 +30,12 @@ import { PREFERRED_SIGNING_ALG } from "../constant";
  * @returns The normalized URL as a string.
  * @hidden
  */
-function removeHashUsernameAndPassword(audience: string): string {
+function normalizeHTU(audience: string): string {
   const cleanedAudience = new URL(audience);
   cleanedAudience.hash = "";
   cleanedAudience.username = "";
   cleanedAudience.password = "";
+  cleanedAudience.search = "";
```

Comment:
Would it be simpler to create a new URL with only protocol + pathname, rather than excluding fields?

Reply:
That would probably be cleaner, and prevent from future bugs like this one. What do you think about 9e1f1fec243379593be346900c0e8711fd4d78aa ?